File: Sistema-asistencia-backend/src/main/java/api/example/apis/Evento.java
package api.example.apis;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Evento {
    private Long id;
    private String nombreEvento;
    private String fechaHoraEntrada;
    private String fechaHoraSalida;
    private int capacidad;
    private String descripcion;
}

File: Sistema-asistencia-backend-laravel/app/Models/Evento.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Evento extends Model
{
    use HasFactory;
    protected $table = 'evento';
    protected $primaryKey = 'ID_Evento';
    protected $fillable = [
        'NombreEvento', 'Capacidad', 'Descripcion', 'FechaHoraEntrada', 'FechaHoraSalida'
    ];
    public function usuarios()
    {
        return $this->belongsToMany(Usuario::class, 'asiste', 'ID_Evento', 'ID_Usuario');
    }
    public function registrosAsistencia()
    {
        return $this->hasMany(RegistroAsistencia::class, 'ID_Evento', 'ID_Evento');
    }
}
